include/circt/Dialect/FIRRTL/FIRRTLExpressions.td

```diff
@@ -1154,6 +1154,23 @@ def ListCreateOp : FIRRTLOp<"list.create", [Pure, SameTypeOperands]> {
   let hasVerifier = 1;
 }
 
+def ListConcatOp : FIRRTLOp<"list.concat", [Pure, SameOperandsAndResultType]> {
+  let summary = "Concatenate multiple lists to produce a new list";
+  let description = [{
+    Produces a value of list type by concatenating the provided lists.
+
+    Example:
+    ```mlir
+    %3 = firrtl.list.concat %0, %1, %2 : !firrtl.list<string>
+    ```
+  }];
+
+  let arguments = (ins Variadic<ListType>:$subLists);
+  let results = (outs ListType:$result);
+
+  let assemblyFormat = "$subLists attr-dict `:` type($result)";
```

Comment:
Might be neat to give this a folder in the future for IR simplification (and use w/IMCP) but that can come later.

Reply:
Could be interesting! I think there are lot of interesting optimizations we could explore on properties.

Reply:
Yeah just thinking about cleaning up the trivial stuff that literally constant-folds.  I'm not sure we're likely to feed this constants at first/soon and yep lots of interesting things for sure! Look forward to it, LMK if want some assistance!

Looks like integer ops have issues filed for adding folders there, but quick check on some internal usage suggests these wouldn't help much just yet anyway.

Anyway stray thought, GLHF!